src/main/scala/firrtl/passes/memlib/ReplaceMemTransform.scala

```diff
@@ -70,19 +73,76 @@ class ConfWriter(filename: String) {
 
 case class ReplSeqMemAnnotation(inputFileName: String, outputConfig: String) extends NoTargetAnnotation
 
+/** Generate conf file for a sequence of [[DefAnnotatedMemory]]
+  * @note file already has its suffix adding by `--replSeqMem`
+  */
+case class MemLibOutConfigFileAnnotation(file: String)
+    extends NoTargetAnnotation
+    with HasAnnotatedMemories
+    with CustomFileEmission {
+  def baseFileName(annotations: AnnotationSeq) = file
+  def suffix = None
+  def getBytes = annotatedMemories.map { m =>
+    require(bitWidth(m.dataType) <= Int.MaxValue)
+    m.maskGran.foreach(x => require(x <= Int.MaxValue))
+    MemConf(
+      m.name,
+      m.depth,
+      bitWidth(m.dataType).toInt,
+      m.readers.length,
+      m.writers.length,
+      m.readwriters.length,
+      m.maskGran.map(_.toInt)
+    ).toString
+  }.mkString("\n").getBytes
+}
+
+trait HasAnnotatedMemories {
```

Comment:
Why make this a trait (especially unsealed) rather than just methods on the class?

More problematic, I don't love the mutability here. Why not have 2 different annotations (shared functionality in a trait perhaps 😉), 1 before the information is evaluated and 1 after?

Reply:
I removed this API and added `private[memlib] case class AnnotatedMemoriesCollectorAnnotation() extends NoTargetAnnotation`, I don't like mutable neither, but I think in this transform we must have a mutable thing to store those `DefAnnotatedMemory`, during walking the entire circuit: originally it was the config writer, in b385574a, it was `HasAnnotatedMemories`, and in 7f82ea10, it is `AnnotatedMemoriesCollectorAnnotation`, I use `insert` and `done` function to make sure it only can be written during walking the circuit at first time, so I think it's safe here.